index.js

```diff
@@ -31,6 +31,29 @@ var isPlainObject = function isPlainObject(obj) {
 	return typeof key === 'undefined' || hasOwn.call(obj, key);
 };
 
+// If name is '__proto__', define it as an own property on target
+var setProperty = function setProperty(target, options) {
+	if (options.name === '__proto__') {
+		Object.defineProperty(target, options.name, {
+			enumerable: true,
+			configurable: true,
+			value: options.newValue,
+			writable: true
+		});
+	} else {
+		target[options.name] = options.newValue;
+	}
+};
+
+// Return a new object instead of __proto__ if '__proto__' is not an own property
+var getProperty = function getProperty(obj, name) {
+	if (name === '__proto__' && !hasOwn.call(obj, name)) {
+		return {};
```

Comment:
should this not be `undefined` in that case?

Reply:
Yes, definitely.